### PR TITLE
Generate native-image binaries for google-java-format

### PR DESCRIPTION
**Pull request by cushon**
And include them in release artifacts.

I tested this on my fork of the repo, and have a demo here: https://github.com/cushon/google-java-format/releases

When downloading the artifacts from the releases page they don't have the executable bit set, and my Mac blocks them because they aren't signed. That can be worked around with the following, but isn't ideal:

```
chmod a+rx google-java-format-darwin
sudo xattr -r -d com.apple.quarantine google-java-format-darwin
```

Progress towards #868, builds on @vorburger's work in #1048.

### CODE REVIEW

**Comment by cushon** [comment]
@vorburger can I get your review of this?

File: .github/workflows/release.yml
Comment (vorburger) [inline .github/workflows/release.yml:89]:
```suggestion
        os: [ubuntu-latest, macos-latest, windows-latest]
```

File: .github/workflows/release.yml
Comment (vorburger) [inline .github/workflows/release.yml:104]: @cushon I don't understand this expression... and it (now) won't work for Windows?

**Review by vorburger — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml
Comment (cushon) [reply .github/workflows/release.yml:104]: It returns the string 'linux' or 'darwin' depending on the current OS, so the platform-specific outputs get uploaded with unique names. The syntax is documented here: https://docs.github.com/en/actions/learn-github-actions/expressions

I agree we'll want to update this for windows, I'm not sure what the cleanest way to do that is yet. I looked around at a few different ways to upload these artifacts to the release and didn't find anything very appealing.

Comment (vorburger) [reply .github/workflows/release.yml:104]: Oh, I think I just got it - the idea is that this will evaluate to the following, right?

* `google-java-format-linux`
* `google-java-format-darwin`

How about we use slightly different suffixes for the released native binaries? Inspired by e.g. `uname -a` output, I would suggest:

* `google-java-format_linux-x86_64`
* `google-java-format_darwin-arm64` (or something like that - what exactly does `uname -a` output on your Mac?)
* `google-java-format_win-x64` (I think that's what that's commonly referred to - but I'm not 100% sure)

Comment (cushon) [reply .github/workflows/release.yml:104]: Those suffixes LGTM, thanks. (My uname output includes `Darwin` and `arm64`.)

Comment (vorburger) [reply .github/workflows/release.yml:104]: > I agree we'll want to update this for windows, I'm not sure what the cleanest way to do that is yet. I looked around at a few different ways to upload these artifacts to the release and didn't find anything very appealing.

Simplest could be to duplicate this (and the "gh release upload") lines x3, with an "if" for penguins, apples, and wins?

**Review by vorburger — APPROVED** [review]
LGTM from me, if you wanted to already merge this as-is?

File: .github/workflows/release.yml
Comment (vorburger) [reply .github/workflows/release.yml:104]: > * `google-java-format_win-x64` (I think that's what that's commonly referred to - but I'm not 100% sure; but I would _NOT_ use "x86_64" à la Linux for this binary, because MS seems to use x86=32bit VS x64=64bit - apparently.)

Just noticed on https://github.com/bazelbuild/bazel/releases that they use "*windows_x86-64" so perhaps we can go for that here as well after all.

Comment (cushon) [reply .github/workflows/release.yml]: Thanks! I found a different way to express that mapping, and updated to use the suggested names.

File: .github/workflows/release.yml
Comment (vorburger) [inline .github/workflows/release.yml:91]: Fancy!! That's a really neat trick - TIL; Thank You!

File: .github/workflows/release.yml
Comment (vorburger) [inline .github/workflows/release.yml:91]:
```suggestion
      SUFFIX: ${{fromJson('{"ubuntu-latest":"linux-x86-64", "macos-latest":"darwin-arm64", "windows-latest":"windows-x86-64"}')[matrix.os]}}
```

File: .github/workflows/release.yml
Comment (vorburger) [inline .github/workflows/release.yml:106]:
```suggestion
        run: cp core/target/google-java-format google-java-format_${{ env.SUFFIX }}
```

File: .github/workflows/release.yml
Comment (vorburger) [inline .github/workflows/release.yml:111]:
```suggestion
        run: gh release upload "v${{ github.event.inputs.version }}" "google-java-format_${{ env.SUFFIX }}"
```

**Review by vorburger — APPROVED** [review]
Proposed changes, you'll also want to rebase this after merging #1060; otherwise LGTM!

**Pull request merged**
